[Bug]: テスト test_retry_002_retry_config_values がリトライ設定の不一致により失敗する

# [Bug]: テスト `test_retry_002_retry_config_values` がリトライ設定の不一致により失敗する

## Current Behavior
結合テスト `tests/test_integration/test_retry.py::TestRetryNormal::test_retry_002_retry_config_values` を実行すると、以下のアサーションエラーが発生して失敗します。

```
>       assert 429 in retry.status_forcelist
E       assert 429 in [500, 502, 503, 504]
E        +  where [500, 502, 503, 504] = Retry(total=3, connect=None, read=None, redirect=None, status=None).status_forcelist

tests/test_integration/test_retry.py:46: AssertionError
```

テストコードは、`urllib3` の標準リトライ機能（`status_forcelist`）にステータスコード `429` (Too Many Requests) が含まれていることを期待しています。

一方、`jquants/client_v2.py` の実装では、429 エラーは標準の `Retry` ストラテジから意図的に除外され、`_request` メソッド内で独自のループ処理として実装されています。

**実装コード (`jquants/client_v2.py`):**
```python
retry_strategy = Retry(
    total=3,
    status_forcelist=[500, 502, 503, 504],  # 429 excluded for custom retry
    # ...
)
```

## Expected Behavior
テストと実装が一致し、全ての結合テストが通過すること。

実装が意図的に 429 を除外しているのであれば、テストコード側を修正して、`status_forcelist` に 429 が**含まれていないこと**を確認するように変更すべきです。

## SDK Version I encountered this issue in
- jquants-api-client version: 0.0.0 (current development version)

## Steps To Reproduce
1. APIキーを設定した環境を用意する（または `jquants-api.toml` を配置）。
2. 以下のコマンドを実行して結合テストを行う。

```bash
poetry run pytest -m integration tests/test_integration/test_retry.py
```

3. `test_retry_002_retry_config_values` が失敗することを確認する。

## Anything else?
### 修正案
`tests/test_integration/test_retry.py` のテストケースを修正し、429 が `status_forcelist` に含まれないことを正とするアサーションに変更することを推奨します。

```python
# 現在
assert 429 in retry.status_forcelist

# 修正案
assert 429 not in retry.status_forcelist
```

---

## 修正設計

### What（入力/出力）

#### 入力
- テストファイル: `tests/test_integration/test_retry.py`
- 対象メソッド: `TestRetryNormal.test_retry_002_retry_config_values`

#### 出力
- `status_forcelist`の検証ロジックを実装に合わせて修正

### Constraint（制約）

1. **変更対象**: テストコードのみ（実装は変更しない）
   - 実装の設計は正しい（429は意図的にカスタム処理）
   
2. **後方互換性**: なし（テストコードのみの変更）

3. **429リトライの設計確認**
   - 公式API: [J-Quants API仕様](https://jpx.gitbook.io/j-quants-ja/)
   - 429は`_request`メソッドでカスタム処理（`retry_on_429`, `retry_wait_seconds`, `retry_max_attempts`）
   - `status_forcelist`はサーバーエラー専用（500, 502, 503, 504）

4. **テスト責務の分離**
   - カスタム429リトライ設定のデフォルト値は`tests/test_client_v2.py`のユニットテストでカバー済み
   - RETRY-002はurllib3 Retry設定の検証に専念

### 修正方針（疑似コード）

```
RETRY-002テストを以下のように修正:

1. status_forcelistの検証:
   - 429が含まれない ことを確認
   - サーバーエラー (500, 502, 503, 504) が含まれる ことを確認
   - 型/順序非依存で比較（set()使用）
```

### 検証観点

| 観点 | 検証内容 |
|------|----------|
| urllib3 Retry設定 | `status_forcelist`に429が**含まれない**ことを確認 |
| urllib3 Retry設定 | `status_forcelist`にサーバーエラー(500-504)が含まれることを確認（型/順序非依存） |
| リトライ共通設定 | `total=3`, `backoff_factor=0.5`, `respect_retry_after_header=True` |

※ カスタム429リトライ設定のデフォルト値は`test_client_v2.py`でテスト済み

--- a/tests/test_integration/test_retry.py
+++ b/tests/test_integration/test_retry.py
@@ -31,7 +31,10 @@ class TestRetryNormal:
         assert isinstance(session1, requests.Session)
 
     def test_retry_002_retry_config_values(self, fresh_client: ClientV2) -> None:
-        """RETRY-002: Retry strategy configuration is correct."""
+        """RETRY-002: Retry strategy configuration is correct.
+
+        429 is handled by custom retry logic in _request(), not by urllib3 Retry.
+        """
         # Access the session to trigger creation
         session = fresh_client._request_session()
 
@@ -39,13 +42,22 @@ class TestRetryNormal:
         adapter = session.get_adapter("https://")
         assert isinstance(adapter, HTTPAdapter)
 
-        # Verify retry configuration
+        # Verify urllib3 Retry configuration
         retry = adapter.max_retries
         assert retry.total == 3
         assert retry.backoff_factor == 0.5
-        assert 429 in retry.status_forcelist
-        assert 500 in retry.status_forcelist
         assert retry.respect_retry_after_header is True
+
+        # 429 is excluded from status_forcelist (handled by custom retry logic)
+        assert 429 not in retry.status_forcelist
+
+        # Server errors are in status_forcelist (handled by urllib3 Retry)
+        assert retry.status_forcelist == [500, 502, 503, 504]
+
+        # Verify custom 429 retry settings (default values)
+        assert fresh_client._retry_on_429 is True
+        assert fresh_client._retry_wait_seconds == 310
+        assert fresh_client._retry_max_attempts == 3
 
 
 class TestRetryRare:

--- a/tests/test_integration/test_retry.py
+++ b/tests/test_integration/test_retry.py
@@ -34,6 +34,7 @@ class TestRetryNormal:
         """RETRY-002: Retry strategy configuration is correct.
 
         429 is handled by custom retry logic in _request(), not by urllib3 Retry.
+        Custom 429 retry defaults are tested in test_client_v2.py unit tests.
         """
         # Access the session to trigger creation
         session = fresh_client._request_session()
@@ -52,12 +53,8 @@ class TestRetryNormal:
         assert 429 not in retry.status_forcelist
 
         # Server errors are in status_forcelist (handled by urllib3 Retry)
-        assert retry.status_forcelist == [500, 502, 503, 504]
-
-        # Verify custom 429 retry settings (default values)
-        assert fresh_client._retry_on_429 is True
-        assert fresh_client._retry_wait_seconds == 310
-        assert fresh_client._retry_max_attempts == 3
+        # Use set() for type/order-independent comparison (urllib3 may use frozenset)
+        assert set(retry.status_forcelist) == {500, 502, 503, 504}
 
 
 class TestRetryRare:

